packages/gatsby/src/utils/page-ssr-module/entry.ts

```diff
@@ -33,12 +36,17 @@ const pageTemplateDetailsMap: Record<
   // @ts-ignore INLINED_TEMPLATE_TO_DETAILS is being "inlined" by bundler
 > = INLINED_TEMPLATE_TO_DETAILS
 
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare const __non_webpack_require__: typeof require
+
 export async function getData({
   pathName,
   graphqlEngine,
+  req,
 }: {
   graphqlEngine: GraphQLEngine
   pathName: string
+  req?: Request
 }): Promise<ISSRData> {
   const potentialPagePath = getPagePathFromPageDataPath(pathName) || pathName
```

Comment:
Addition here - `getData` in page engine now accept `req` (which will be used if path is SSR to provide headers and query params).